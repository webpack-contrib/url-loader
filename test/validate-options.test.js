import loader from '../src';

it('validate options', async () => {
  const validate = (options) =>
    loader.call(
      Object.assign(
        {},
        {
          resourcePath: 'image.png',
          query: options,
          emitFile: () => {},
        }
      ),
      'context'
    );

  expect(() => validate()).not.toThrow();

  // The `fallback` loader can have any optsions so we use `additionalProperties: false` to avoid problems.
  expect(() => validate({ unknown: 'unknown' })).not.toThrow();

  expect(() => validate({ limit: 8192 })).not.toThrow();
  expect(() => validate({ limit: true })).not.toThrow();
  expect(() => validate({ limit: '8192' })).not.toThrow();
  expect(() => validate({ limit: [] })).toThrowErrorMatchingSnapshot();

  expect(() => validate({ mimetype: 'image/png' })).not.toThrow();
  expect(() => validate({ mimetype: true })).toThrowErrorMatchingSnapshot();

  expect(() => validate({ fallback: 'custom-loader' })).not.toThrow();
  expect(() =>
    validate({ fallback: { loader: 'custom-loader' } })
  ).not.toThrow();
  expect(() =>
    validate({
      fallback: { loader: 'custom-loader', options: { unknown: 'unknown' } },
    })
  ).not.toThrow();
  expect(() =>
    validate({
      fallback: { loader: 'custom-loader', options: 'string' },
    })
  ).not.toThrow();
  expect(() => validate({ fallback: true })).toThrowErrorMatchingSnapshot();
});
