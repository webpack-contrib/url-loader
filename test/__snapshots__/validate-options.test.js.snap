// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"URL Loader Invalid Options

options.limit should be boolean,number,string
"
`;

exports[`validation 2`] = `
"URL Loader Invalid Options

options.mimetype should be string
"
`;

exports[`validation 3`] = `
"URL Loader Invalid Options

options.fallback should be string
options.fallback should be object
options.fallback should match some schema in anyOf
"
`;
